File: 1502. Can Make Arithmetic Progression From Sequence.py
class Solution:
    def canMakeArithmeticProgression(self, arr: List[int]) -> bool:
        if arr != sorted(arr):
            arr = sorted(arr)
        print(arr)
        
        d = arr[0] - arr[1]
        
        for i in range(len(arr)-1):
            d_dash = arr[i] - arr[i+1]
            if d != d_dash:
                return False
        return True

File: 1502. Can Make Arithmetic Progression From Sequence.cpp
#include<algorithm>
class Solution {
public:
    bool canMakeArithmeticProgression(vector<int>& arr) {
        sort(arr.begin(), arr.end());
        //bool ans = true;
        int d = arr[0] - arr[1];
        for(int i = 0; i < arr.size() - 1; i++){
            if((arr[i] - arr[i+1]) != d){
                return false;
            }
        }return true;
    }
};